clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -412,3 +437,48 @@ func GetNumObjectsMatchingSelector(c dynamic.Interface, namespace string, resour
 	err := client.RetryWithExponentialBackOff(client.RetryFunction(listFunc))
 	return numObjects, err
 }
+
+// The pod can only schedule onto nodes that satisfy requirements in NodeAffinity.
+func podMatchesNodeAffinity(affinity *corev1.Affinity, node corev1.Node) (bool, error) {
```

Comment:
For performance reasons, we don't want to pass corev1.Node (because that is resulting in making a copy of the object).
Instead you want to pass *corev1.Node

Reply:
Same everywhere below